File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {

        int[][] map1 = new int[n][n];
        int[][] map2 = new int[n][n];
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            for (int j = 1; j <= n; j++) {
                map1[i][n-j] = arr1[i] % 2;
                map2[i][n-j] = arr2[i] % 2;
                arr1[i] /= 2;
                arr2[i] /= 2;
            }
        }

        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if(map1[i][j] == 1 || map2[i][j] == 1) {
                    sb.append("#");
                } else {
                    sb.append(" ");
                }
            }
            answer[i] = sb.toString();
            sb = new StringBuffer();
        }
        return answer;
    }
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    map1 = [[0] * n for _ in range(n)]
    map2 = [[0] * n for _ in range(n)]
    
    for i in range(n):
        for j in range(n):
            map1[i][n-j-1] = arr1[i] % 2
            map2[i][n-j-1] = arr2[i] % 2
            arr1[i] //= 2
            arr2[i] //= 2
    
    answer = ["" for _ in range(n)]
    for i in range(n):
        for j in range(n):
            if(map1[i][j] == 0 and map2[i][j] == 0):
                answer[i] += " "
            else:
                answer[i] += "#"
        
    
    
    return answer